pkg/controller/secretgenerator/secretgenerator_controller.go

```diff
@@ -138,3 +175,80 @@ func (r *ReconcileSecretGenerator) Reconcile(request reconcile.Request) (reconci
 	// Don't requeue if secret is created successfully
 	return reconcile.Result{}, err
 }
+
+// getRoute returns an OpenShift route object. It waits until the .spec.host value exists to avoid possible race conditions, fails otherwise.
+func (r *ReconcileSecretGenerator) getRoute(name string, namespace string) (*routev1.Route, error) {
+	oauthClientRoute := &routev1.Route{}
```

Comment:
Replace s/oauthClientRoute/route to make it more generic (it can return any route, not only the OAuth client route).